lib/agenda/index.ts

```diff
@@ -28,6 +28,23 @@ import { stop } from './stop';
 import { findAndLockNextJob } from './find-and-lock-next-job';
 import { Job } from '../job';
 
+export interface AgendaConfig {
```

Comment:
looks good, but you could also "split" it up like I did in the hokify fork. E.g. right now it doesn't make sense if a use provides "db" and "mongo".. by using a construct like this https://github.com/hokify/agenda/blob/c5e21e98b92ef1ba77dab36f68a331c4b493b382/src/index.ts#L85 you could achive a higher level of accuracy

see also https://github.com/hokify/agenda/blob/master/src/types/DbOptions.ts